### PR TITLE
Adding rewrites involving kronecker product

### PR DESCRIPTION
**Pull request by tanish1729**
## Description

Adds the following rewrites :
- diag(kron(a,b)) -> outer(diag(a), diag(b))
- slogdet(kron(a,b)) -> prod(sign(a), sign(b)), sum(logdet(a), logdet(b))

## Related Issue

- [ ] Related to #573 

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pytensor/tensor/rewriting/linalg.py [hunk 1]
```diff
@@ -611,3 +613,20 @@ def rewrite_inv_inv(fgraph, node):
     ):
         return None
     return [potential_inner_inv.inputs[0]]
+
+
+@register_canonicalize
+@register_stabilize
+@node_rewriter([ExtractDiag])
+def rewrite_diag_kronecker(fgraph, node):
```
Comment (jessegrabowski) [inline pytensor/tensor/rewriting/linalg.py:621]: What about diagonals that result from multiplication with eye?

Comment (jessegrabowski) [reply pytensor/tensor/rewriting/linalg.py:621]: Also needs a docstring

**Review by jessegrabowski — CHANGES_REQUESTED** [review]

File: pytensor/tensor/rewriting/linalg.py (hunk 1 above)
Comment (tanish1729) [reply pytensor/tensor/rewriting/linalg.py:621]: this rewrite does not deal with the creation of a diagonal matrix. the input should be the kron product and diag is the outer op

**Review by jessegrabowski — APPROVED** [review]
Small typo then looks good to me

**Pull request merged**